app/models/shipit/provisioning_handler/unregistered_provisioning_handler.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module Shipit
+  module ProvisioningHandler
+    class UnregisteredProvisioningHandler < Shipit::ProvisioningHandler::Base
+      def up
+        lock_and_prevent_transition
+      end
+
+      def down
+        lock_and_prevent_transition
+      end
+
+      private
+
+      def lock_and_prevent_transition
+        stack.lock(
+          "Failed to find a provisioning handler named '#{stack.provisioning_handler_name}' in the " \
+          "ProvisioningHandler registry. Have you registered it via Provisioning::Handler.register?",
+          Shipit::AnonymousUser.new
+        )
+        throw(:halt)
```

Comment:
Two things here:

  - The only `catch` I see are in `test/`. 
  - `catch` is really not idiomatic at all, I'm really unsure what you are trying to do here.

Reply:
I believe it's used internally by the [state_machines](https://github.com/state-machines/state_machines-activerecord) gem as a transition control mechanism. This is intended to prevent the `ReviewStack`'s `provision_status` state machine from transitioning to a "provisioned" state if the stack couldn't provision.

Reply:
I see. I didn't know about that, and it's not really documented in their README, but OK. Can you just add an inline comment explaining it abort the state transition?

Reply:
Addressed by: 0af0c0b7